product_attribute_global_item_code/product.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+###############################################################################
+#
+#   Copyright (C) 2015 Akretion (http://www.akretion.com). All Rights Reserved
+#   @author Abdessamad HILALI <abdessamad.hilali@akretion.com>
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Affero General Public License as
+#   published by the Free Software Foundation, either version 3 of the
+#   License, or (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from openerp import fields, models, api, _
+from openerp.exceptions import Warning
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    base_code = fields.Char('Base Code',
+                            help="this field is used as a base "
+                            "for automatically create "
+                            "Internal Reference(default_code)")
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    use_internal_ref = fields.Boolean('Manual Internal Reference',
+                                      default=False)
+    manual_default_code = fields.Char()
+    default_code = fields.Char(compute="_compute_default_code",
+                               inverse="_set_manual_default_code",
+                               store=True)
+
+    @api.multi
+    def _build_code(self, vals):
+        self.ensure_one()
```

Comment:
Put a doc string to indicate that this is method to customize: "inherit this method to customize your own behavior"